test/raven.test.js

```diff
@@ -2216,6 +2251,193 @@ describe('Raven (public API)', function() {
                 { message: 'lol', timestamp: 0.1 }
             ]);
         });
+
+        describe('breadcrumbCallback', function() {
+            it('should filter the breadcrumb if it returns false', function() {
+                Raven.setBreadcrumbCallback(function() {
+                    return false;
+                });
+
+                Raven.captureBreadcrumb({
+                    type: 'http',
+                    data: {
+                        url: 'http://example.org/api/0/auth/',
+                        status_code: 200
+                    }
+                });
+
+                assert.strictEqual(Raven._breadcrumbs.length, 0);
+            });
+
+            it('should not filter the breadcrumb if callback returns undefined', function() {
+                Raven.setBreadcrumbCallback(function() {});
+
+                Raven.captureBreadcrumb({
+                    type: 'http',
+                    data: {
+                        url: 'http://example.org/api/0/auth/',
+                        status_code: 200
+                    }
+                });
+
+                assert.deepEqual(Raven._breadcrumbs, [{
+                    type: 'http',
+                    timestamp: 0.1,
+                    data: {
+                        url: 'http://example.org/api/0/auth/',
+                        status_code: 200
+                    }
+                }]);
+            });
+
+            it('should mutate the breadcrumb if it returns an object', function() {
```

Comment:
"replace" is a better descriptor than "mutate" – you can mutate the input crumb and not return anything, different from replacing the value